doc/code-tests.md

```diff
@@ -0,0 +1,185 @@
+# Code tests
+
+For its front end, Numberscope uses [`vitest`](https://vitest.dev/) for unit
+tests and low-level integration tests that run without a browser, and
+[`Playwright`](https://playwright.dev/) for higher-level integration and
+end-to-end integration tests that run in a browser
+
+### Running tests
+
+The package manager is set up with scripts that make basic running of all
+tests very convenient. To run all unit (vitest) tests, execute
+`npm run test:unit`. To run all end-to-end (Playwright) tests, use
+`npm run test:e2e`.
+
+A couple of additional points about the end-to-end tests:
+
+-   They result in accessing a considerable amount of (sequence) data over the
+    internet. Therefore, if you run them when your internet connection is poor
+    or inactive, some of the tests will fail.
+
+-   By default, they mostly run in a [Docker](https://www.docker.com/)
+    container, which we have found maximizes the reproducibility of the tests.
+    You can also run them directly in your ambient system with
+    `npm run e2e:test:local` or `npm run e2e:test:ui` for the graphical front
```

Comment:
These commands don't seem to be correct.  Are they supposed to be `npm run test:e2e:cmd` and `npm run test:e2e:ui`?